### PR TITLE
fix(insights): links were pointing to old dashboards

### PR DESCRIPTION
**Pull request by bcoe**
We missed migrating some links with the new Insights menu.

### CODE REVIEW

File: src/components/pageGrid.tsx [hunk 1]
```diff
@@ -20,9 +20,9 @@ export function PageGrid({header, exclude}: Props) {
   }
 
   const children: DocNode[] = parentNode.frontmatter.next_steps?.length
-    ? (parentNode.frontmatter.next_steps
+    ? parentNode.frontmatter.next_steps
```
Comment (bcoe) [inline src/components/pageGrid.tsx]: This change was made by running `eslint:fix` and `prettier:fix`.

File: docs/platforms/go/common/tracing/instrumentation/caches-module.mdx [hunk 2]
```diff
@@ -91,4 +91,4 @@ if parentSpan != nil {
 }
 ```
 
-You should now have the right spans in place. Head over to the [Cache dashboard](https://sentry.io/orgredirect/organizations/:orgslug/performance/caches/) to see how your cache is performing.
+You should now have the right spans in place. Head over to the [Cache dashboard](https://sentry.io/orgredirect/organizations/:orgslug/insights/backend/caches) to see how your cache is performing.
```
Comment (DominikB2014) [inline docs/platforms/go/common/tracing/instrumentation/caches-module.mdx:94]:
```suggestion
You should now have the right spans in place. Head over to the [Cache dashboard](https://sentry.io/orgredirect/organizations/:orgslug/insights/backend/caches/) to see how your cache is performing.
```

Sometimes there's weird behaviour when we don't prepend with a slash, so I suggest we add one here

File: docs/product/insights/index.mdx [hunk 3]
```diff
@@ -6,7 +6,7 @@ description: Sentry's Insights for frontend, backend, mobile, and AI, provide cu
...
+Sentry’s frontend, backend, mobile, and AI performance insights (all under the [**Insights**](https://sentry.io/orgredirect/organizations/:orgslug/insights/backend) heading in Sentry), give you a clear summary of the telemetry and events impacting the performance of individual parts of your application. Each section shows a high-level overview of your application's health with widgets for "Best Page Opportunities" (improvement that will be most helpful for boosting your performance score), most time-consuming events, and so on. You can also drill deeper into specific workflows to see specific traces, errors, and releases. This helps you identify potential problems and spot trends, fix bugs, and ultimately boost application performance.
```
Comment (DominikB2014) [inline docs/product/insights/index.mdx:9]:
```suggestion
Sentry’s frontend, backend, mobile, and AI performance insights (all under the [**Insights**](https://sentry.io/orgredirect/organizations/:orgslug/insights/backend/) heading in Sentry), give you a clear summary of the telemetry and events impacting the performance of individual parts of your application. Each section shows a high-level overview of your application's health with widgets for "Best Page Opportunities" (improvement that will be most helpful for boosting your performance score), most time-consuming events, and so on. You can also drill deeper into specific workflows to see specific traces, errors, and releases. This helps you identify potential problems and spot trends, fix bugs, and ultimately boost application performance.
```

**Review by DominikB2014 — APPROVED** [review]
Lgtm, but a couple small nits

**Pull request merged**